internal/httpx/httpx.go

```diff
@@ -142,14 +143,35 @@ func (c *apiClient) newRequestWithJSONBody(
 	return request, nil
 }
 
+// joinURLPath joins the BaseURL and the URLPath, allows httpx to access the URL
+func (c *apiClient) joinURLPath(origPath string, newPath string) string {
+
+	if origPath != "" && newPath != "" {
+		if !strings.HasSuffix(origPath, "/") && !strings.HasPrefix(newPath, "/") {
+			origPath += "/"
+		}
+
+		if strings.HasSuffix(origPath, "/") && strings.HasPrefix(newPath, "/") {
+			newPath = newPath[1:]
+		}
+		return origPath + newPath
+	}
+
+	if origPath != "" && newPath == "" {
+		return origPath
+	}
+
+	return ""
```

Comment:
```suggestion
	if !strings.HasSuffix(origPath, "/")  {
		origPath += "/"
	}
	if strings.HasPrefix(newPath, "/") {
		newPath = newPath[1:]
	}
	return origPath + newPath
```

Would this (simpler) implementation work? Does it look correct?

(If we can achieve the same objective with less code, it's probably better in terms of future maintenance.)

Reply:
Also, please, modify the documentation of `httpx` to ensure we document what happens when the baseURL has a path and also the resource URL has a path. Thank you!